### PR TITLE
Added information about runtime version

### LINKED ISSUE #797: Add note about runtime <> metadata correspondence
I've had a few wallet teams ask if they need to fetch the (rather large) metadata every time they make a transaction. Can we add a note to the metadata documentation page that the metadata corresponds to a runtime spec number? It should clarify that:

- Spec version => unique metadata
- Transaction version => unique call mapping

### PR DESCRIPTION
**Pull request by jimmychu0807**
Close #797 

@danforbes  Please take a look to see if the content fit in the metadata section, or suggesting another place that the content could fit better.

@joepetrowski  Please take a look to see I cover the info you want to let the audience know.

Thank you all.

### CODE REVIEW

File: docs/knowledgebase/runtime/metadata.md [hunk 1]
```diff
@@ -443,6 +443,34 @@ These are errors that could occur during the submission or execution of an extri
...
+- `transaction_version` is the version of the extrinsics interface. This number is updated when the chain extrinsic interface and ID (both module and dispatch ID) are changed. This means this number is updated if extrinsic parameters and their types have been changed; extrinsics or modules are being removed; or module order in `construct_runtime!` macro and extrinsic order in a module has been changed (causing a change in their module/dispatch ID). If this number is updated, then the `spec_version` must also be updated. 
+- `apis` is a list of supported API "features" along with their versions.
+
+When submitting an extrinsic to a chain, one could query the `transaction_version` of the runtime and its metadata to learn the chain extrinsic interface. When submitting another extrinsic at a latter point in time, one could query the chain `transaction_version` first to check if the chain extrinsic interface is staying the same. If the `trasaction_version` has been updated, then the runtime metadata must be queried again to get the latest chain extrinsic interface.
```
Comment (joepetrowski) [inline docs/knowledgebase/runtime/metadata.md:470]: Would be good to add that metadata only changes when `spec_version` changes.

File: docs/knowledgebase/runtime/metadata.md [hunk 2]
```diff
@@ -443,6 +443,51 @@ These are errors that could occur during the submission or execution of an extri
...
+- `apis` is a list of supported API "features" along with their versions.
+
+When submitting an extrinsic to a chain, one could query the `transaction_version` of the runtime
+and its metadata to learn the chain extrinsic interface. When submitting another extrinsic at a
+latter point in time, one could query the chain `transaction_version` first to check if the chain
+extrinsic interface is staying the same. If the `trasaction_version` has been updated, then the
+runtime metadata must be queried again to get the latest chain extrinsic interface.
+
+Runtime version structure can be queried using the chain RPC call
+[`state.getMetadata()`](https://polkadot.js.org/docs/substrate/rpc#getmetadataat-blockhash-metadata).
```
Comment (danforbes) [inline docs/knowledgebase/runtime/metadata.md:489]: This feels like a duplication of the information that is already provided here https://substrate.dev/docs/en/knowledgebase/advanced/executor#runtime-versioning I think it's best to keep this information in one place.

Comment (joepetrowski) [reply docs/knowledgebase/runtime/metadata.md:489]: Didn't know about this. In any case, it needs to mention that there is one-to-one correspondence between `spec_version` and the metadata.

Practically speaking: If you have the metadata for runtime `n`, then as long as `spec_version == n` then you would not need to fetch the metadata.

Comment (jimmychu0807) [reply docs/knowledgebase/runtime/metadata.md:489]: hmmm... we need to revisit the search functionality.

**Comment by jimmychu0807** [comment]
@danforbes @joepetrowski 
I have merged the content in `advanced/executor` page.

**Comment by danforbes** [comment]
Marking this as a draft because I want to move the runtime version stuff out of the executor docs. Will do this ASAP.

**Review by danforbes — APPROVED** [review]
@jimmychu0807, @joepetrowski please make sure to review the changes I made under `Accessing the Runtime Version` in the upgrades doc. This replaces some of the original text Jimmy had provided so please make sure it serves the same purpose.

**Comment by jimmychu0807** [comment]
👍  Very cool. Look good to me on reorganising the information into the `update` doc.

**Comment by jimmychu0807** [comment]
I am merging this PR.
@joepetrowski If you find anything we miss, let us know and we will add it back.

**Pull request merged**